submitter/relayer/relayer.go

```diff
@@ -293,12 +308,85 @@ func (rl *Relayer) calculateBumpedFee(ckptInfo *types.CheckpointInfo) btcutil.Am
 	// Calculate the recommended fee using ResubmitFeeMultiplier
 	bumpedFee := ckptInfo.Tx2.Fee.MulF64(rl.config.ResubmitFeeMultiplier)
 
+	if previousFailure == nil ||
+		(!errors.Is(previousFailure, ErrInsufficientFee) &&
+			!errors.Is(previousFailure, ErrInsufficientFeerate) &&
+			!errors.Is(previousFailure, ErrFeeIncrementTooSmall)) {
+		if bumpedFee < requiredFee {
+			bumpedFee = requiredFee
+		}
+
+		return bumpedFee, nil
+	}
+
+	txID := ckptInfo.Tx2.TxID.String()
+	mempoolEntry, err := rl.BTCWallet.GetMempoolEntry(txID)
+	if err != nil {
+		return 0, fmt.Errorf("failed to get mempool entry for %s: %w", txID, err)
+	}
+
+	// Calculate adjustments based on error type and available mempool data
+	switch {
+	case errors.Is(previousFailure, ErrInsufficientFee):
+		// Get the total fees of original tx + descendants
+		originalTotalFees := btcutil.Amount(mempoolEntry.DescendantFees)
```

Comment:
maybe each case can be separate function ie:
```go
case errors.Is(previousFailure, ErrInsufficientFee):
  fee  := handleInsufficientFee(..)
case errors.Is(previousFailure, ErrInsufficientFeerate):
  fee := handeInsufficientFeerate(..)
...
```

It will be much easier for the reader to distinguish the case at first glance, also each function can have a bit of doc how is the calculation fixes the error.